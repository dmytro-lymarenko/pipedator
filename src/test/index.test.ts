import { test } from './index';

describe('test()', () => {
	it('should invoke provided predicate when succeeding', () => {
		const isValid = jest.fn(() => true);

		test(isValid, 'Test').validate(1);

		expect(isValid.mock.calls.length).toBe(1);
		expect(isValid.mock.calls[0][0]).toBe(1);
	});

	it('should invoke provided predicate when failing', () => {
		const isValid = jest.fn(() => false);

		test(isValid, 'Test').validate(1);

		expect(isValid.mock.calls.length).toBe(1);
		expect(isValid.mock.calls[0][0]).toBe(1);
	});

	it('should validate when provided predicate return true', () => {
		expect(test(() => true, 'Test').validate(1)).toBeNull();
	});

	it('should fail when provided predicate return false', () => {
		expect(test(() => false, 'Test').validate(1)).toMatchSnapshot();
	});
});
