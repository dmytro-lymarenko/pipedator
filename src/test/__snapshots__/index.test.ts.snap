// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test() should fail when provided predicate return false 1`] = `
Object {
  "message": "Test",
  "path": Array [],
  "type": "single",
  "value": 1,
}
`;
